### PR TITLE
Revert delete-by-query deprecations

### PR DESCRIPTION
**Pull request by mikemccand**
Un-deprecate delete-by-query since we will not remove it for 2.0 (#10288), but rather hopefully soon replace it with a less trappy implementation (#7052).

### CODE REVIEW

File: src/main/java/org/elasticsearch/action/deletebyquery/IndexDeleteByQueryResponse.java [hunk 1]
```diff
@@ -113,4 +109,4 @@ public void writeTo(StreamOutput out) throws IOException {
             failure.writeTo(out);
         }
     }
-}
+}
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/action/deletebyquery/IndexDeleteByQueryResponse.java]: unintentional removal of newline?

File: src/main/java/org/elasticsearch/action/deletebyquery/ShardDeleteByQueryResponse.java [hunk 2]
```diff
@@ -43,4 +39,4 @@ public void readFrom(StreamInput in) throws IOException {
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
     }
-}
+}
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/action/deletebyquery/ShardDeleteByQueryResponse.java]: ditto?

File: src/main/java/org/elasticsearch/action/deletebyquery/TransportShardDeleteByQueryAction.java [hunk 3]
```diff
@@ -45,10 +45,8 @@
 import org.elasticsearch.transport.TransportService;
 
 /**
- * @deprecated Delete by query will be removed in 2.0.  Instead, use the scroll/scan API to find all matching IDs and then issue a bulk
- * request to delete them.
+ *
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/action/deletebyquery/TransportShardDeleteByQueryAction.java]: Let's just remove an empty javadoc block?

File: src/main/java/org/elasticsearch/index/engine/DeleteByQueryFailedEngineException.java [hunk 4]
```diff
@@ -22,13 +22,11 @@
...
+ *
  */
-@Deprecated
 public class DeleteByQueryFailedEngineException extends EngineException {
 
     public DeleteByQueryFailedEngineException(ShardId shardId, Engine.DeleteByQuery deleteByQuery, Throwable cause) {
         super(shardId, "Delete by query failed for [" + deleteByQuery.query() + "]", cause);
     }
-}
+}
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/index/engine/DeleteByQueryFailedEngineException.java]: another newline removal

File: src/main/java/org/elasticsearch/rest/action/deletebyquery/RestDeleteByQueryAction.java [hunk 5]
```diff
@@ -41,10 +41,8 @@
 import static org.elasticsearch.rest.RestRequest.Method.DELETE;
 
 /**
- * @deprecated Delete by query will be removed in 2.0.  Instead, use the scroll/scan API to find all matching IDs and then issue a bulk
- * request to delete them.
+ *
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/rest/action/deletebyquery/RestDeleteByQueryAction.java]: ditto on removing empty comment?

**Comment by rjernst** [comment]
LGTM, just a couple very minor requests.

**Comment by mikemccand** [comment]
Thanks @rjernst, I added the missing newlines, removed the empty javadocs...

**Pull request closed without merging**